scipy/integrate/tests/test__quad_vec.py

```diff
@@ -207,3 +206,38 @@ def f(x):
     for p in interval_sets:
         j = np.searchsorted(sorted(points), tuple(p))
         assert np.all(j == j[0])
+
+def test_result_object():
+        # Check that result object contains attributes 'success', 'status',
+        # 'neval', 'intervals', 'integrals', 'errors', 'message'.
+        # During the `full_output` deprecation period, also check
+        # that specifying `full_output` produces a warning and that values
+        # are the same whether `full_output` is True, False, or unspecified.
+        def func(x):
+            return x**2 + 1
+
+        res = quad_vec(func, 0, 4)
+        with np.testing.assert_warns(DeprecationWarning,
+                                     match="'full output"):
+            res2 = quad_vec(func, 0, 4, full_output=False)
+        with np.testing.assert_warns(DeprecationWarning,
+                                     match="'full output"):
+            res3 = quad_vec(func, 0, 4, full_output=True)
+
+        assert_equal(res, res2)
+        assert res.success == res2.success
+        assert res.status == res2.status
+        assert res.neval == res2.neval
+        assert_equal(res.intervals, res2.intervals)
```

Comment:
I wouldn't mind using `assert_equal` throughout for consistency, but this isn't enough to re-run CI.